src/Kiota.Builder/Writers/CSharp/CodeClassDeclarationWriter.cs

```diff
@@ -7,6 +7,7 @@ namespace Kiota.Builder.Writers.CSharp;
 public class CodeClassDeclarationWriter : BaseElementWriter<ClassDeclaration, CSharpConventionService>
 {
     public static string AutoGenerationHeader => "// <auto-generated/>";
+    public static string GeneratedCodeAttribute { get; } = $"[global::System.CodeDom.Compiler.GeneratedCode(\"Kiota\", \"{typeof(CodeClassDeclarationWriter).Assembly.GetName().Version}\")]";
```

Comment:
```suggestion
    public static string GeneratedCodeAttribute { get; } = $"[global::System.CodeDom.Compiler.GeneratedCode(\"Kiota\", \"{Kiota.Generated.KiotaVersion.Current()}\")]";
```
You should be able to do something like this instead...

Reply:
I'm not sure that would work - `Kiota.Generated.KiotaVersion.Current()` isn't in the scope of the code writer so the code won't compile, and emitting it as a literal won't work because the attribute has to take a compile-time static value.

Reply:
Odd that it's not available as it's in use https://github.com/microsoft/kiota/blob/59c637ff8b738706074ee4bdc5a33477c23879ff/src/Kiota.Builder/KiotaBuilder.cs#L145

Reply:
Hmm. Maybe I'll try again. I got the squiggle in VS and didn't go any further.

Reply:
Done - I get squiggles in Visual Studio and Visual Studio Code for a compiler warning, but it builds on the CLI.

Reply:
yes, it's a common issue with source generators. Often rebuilding makes it go away. Hopefully the dotnet team will solve it at some point. Thanks for making the changes. Resolving.